### PR TITLE
Basic streamlit app demo

### LINKED ISSUE #1: [UCL-35] Functional test: ensure the hello_world app runs on the GAE
* [X] will need to configure the proxies
* [X] will need a default 'command' to run

waiting on the pull request re streamlit to be closed

<sub>From [SyncLinear.com](https://synclinear.com) | [UCL-35](https://linear.app/drstevok/issue/UCL-35/functional-test-ensure-the-hello-world-app-runs-on-the-gae)</sub>

### PR DESCRIPTION
**Pull request by docsteveharris**
- now stand-up dev services for postgres and synthetic data generation so you can work locally with live data
- validate data on import from the IDS using pandera
- provide an example of basic authentication

Please close and merge #2 before doing this one.
They have the same history so that should work OK.

Fixes GTD-17

### CODE REVIEW

**Comment by JenHunter89** [comment]
**On compose_w_fastapi branch:**

- Devcontainer seems to open with no issues but get the below error trying to run app.py:
- `python web/app.py
Traceback (most recent call last):
  File "/workspaces/devcon2/web/app.py", line 1, in <module>
    from flask import Flask, jsonify
ModuleNotFoundError: No module named 'flask'`

Is this an issue with the requirements?

**Comment by JenHunter89** [comment]
**On streamlit_mwe branch**
- Devcontainer opens with no issues
- When trying to run app.py get error:
- `  File "/workspaces/devcon2/web/app.py", line 32, in create_engine_ids
    f"postgresql+psycopg2://{st.secrets.emap_ids.username}:"
  File "/usr/local/lib/python3.10/site-packages/streamlit/runtime/secrets.py", line 296, in __getattr__
    raise AttributeError(_missing_attr_error_message(key))
AttributeError: st.secrets has no attribute "emap_ids". Did you forget to add it to secrets.toml or the app settings on Streamlit Cloud? More info: https://docs.streamlit.io/streamlit-cloud/get-started/deploy-an-app/connect-to-data-sources/secrets-management
vscode ➜ /workspaces/devcon2 (steve/streamlit-mwe) $ `

I've tried removing 'emap_ids' from the attribute, and changing attribute names in secret file to match (not all do e.g. app.py refers to 'username', secrets refers to 'user', but not sure if this matters?). This hasn't fixed it, still getting an error `no attribute`

**Comment by docsteveharris** [comment]
> **On compose_w_fastapi branch:**
> 
> * Devcontainer seems to open with no issues but get the below error trying to run app.py:
> * `python web/app.py Traceback (most recent call last): File "/workspaces/devcon2/web/app.py", line 1, in <module> from flask import Flask, jsonify ModuleNotFoundError: No module named 'flask'`
> 
> Is this an issue with the requirements?

@JenHunter89 the branches _should_ work. Flask was in the requirements file for the `compose_w_fastapi branch`
see https://github.com/HYLODE/devcon2/blob/7f653cf1b8c903f8723258599c4346e5c1591377/web/requirements.txt#L1

but removed when we got to the `streamlit-mwe` branch

see https://github.com/HYLODE/devcon2/blob/0171745540acd00e60644bd903ae75d1f5f06464/web/app.py#L11 where I have removed flask

**Comment by docsteveharris** [comment]
> **On streamlit_mwe branch**
> 
> * Devcontainer opens with no issues
> * When trying to run app.py get error:
> * ` File "/workspaces/devcon2/web/app.py", line 32, in create_engine_ids f"postgresql+psycopg2://{st.secrets.emap_ids.username}:" File "/usr/local/lib/python3.10/site-packages/streamlit/runtime/secrets.py", line 296, in __getattr__ raise AttributeError(_missing_attr_error_message(key)) AttributeError: st.secrets has no attribute "emap_ids". Did you forget to add it to secrets.toml or the app settings on Streamlit Cloud? More info: https://docs.streamlit.io/streamlit-cloud/get-started/deploy-an-app/connect-to-data-sources/secrets-management vscode ➜ /workspaces/devcon2 (steve/streamlit-mwe) $`
> 
> I've tried removing 'emap_ids' from the attribute, and changing attribute names in secret file to match (not all do e.g. app.py refers to 'username', secrets refers to 'user', but not sure if this matters?). This hasn't fixed it, still getting an error `no attribute`

Sorry. My bad. This should be fixed in https://github.com/HYLODE/devcon2/pull/4/commits/a2a1b6dde2226a38ccaf79f5270f848ab0b00f9d

File: readme.md [hunk 1]
```diff
@@ -12,14 +12,20 @@
...
 Assumes you have VScode installed with the [shell command `code` installed in your path](https://code.visualstudio.com/docs/setup/mac#_launching-from-the-command-line).
 
+### Setup environment variables and secrets
+
 Open a terminal and enter the following commands
 
 ```bash
 cd devcon2
 cp .env.example .env
+cp web/.streamlit/secrets.toml.example web/.streamlit/secrets.toml
```
Comment (docsteveharris) [inline readme.md:22]: @JenHunter89 : not sure if you saw this line when you were doing the review

File: web/.streamlit/secrets.toml.example [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# You need to copy this as below
+# cp .streamlit/secrets.toml.example .streamlit/secrets.toml
+
+password = "notverysecret"
+
+[emap_ids]
+host = "db"
+port = 5432
+dbname = "demo"
```
Comment (dan-hanlon) [inline web/.streamlit/secrets.toml.example]: Had to change this to `database` as per the reference in `app.py`

```suggestion
database = "demo"
```

File: web/.streamlit/secrets.toml.example [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+user = "postgres"
```
Comment (dan-hanlon) [inline web/.streamlit/secrets.toml.example]: Same as above, had to change to `username`

```suggestion
username = "postgres"
```

**Pull request merged**